### ISSUE TITLE
Curator breaks under high rates of node changes

### ISSUE THREAD

**Issue by lorenz** [monogon-dev/monogon#260]
```
root.role.controlplane.launcher.curator.listener.rpc I0803 17:36:43.887279 impl_leader_curator.go:115] Span 1777f01f7b7cec9e: etcd watch failed (initial fetch): node ID mismatch (etcd key: "metropolis-f051eaf799695c2bc855d5a6e156b2d1", value: "metropolis-b4ec87d6e5d318e6cd667c9558565962")
```

**Comment by jscissr** [comment]
A bug which could cause this was fixed in d6a88022d0c8ffdd2b938f161b8825148762b099.

**Comment by q3k** [comment]
Yeah, this was likely the bug:

  1. https://github.com/monogon-dev/monogon/blob/main/metropolis/node/core/curator/state_node.go#L223 Here, we instantiate an etcdPrefix as a package-level variable.
  2. https://github.com/monogon-dev/monogon/blob/main/metropolis/node/core/curator/state_node.go#L397 Here, we use it concurrently from nodeLoad/nodeSave calls.

This means that a race between node{Load,Save} calls (either same function, or different calls) could've lead to a data race and thus etcd corruption.
